ggml/src/ggml-sycl/dpct/helper.hpp

```diff
@@ -1686,26 +1709,18 @@ namespace dpct
 
     namespace detail
     {
-        template <class Ta, class Tb, class Tc, class Ts>
-        inline void gemm_impl(sycl::queue &q, oneapi::mkl::transpose a_trans,
-                              oneapi::mkl::transpose b_trans, int m, int n, int k,
-                              const void *alpha, const void *a, int lda, const void *b,
-                              int ldb, const void *beta, void *c, int ldc)
-        {
-            Ts alpha_value = dpct::get_value(reinterpret_cast<const Ts *>(alpha), q);
-            Ts beta_value = dpct::get_value(reinterpret_cast<const Ts *>(beta), q);
-            auto data_a = get_memory<const Ta>(a);
-            auto data_b = get_memory<const Tb>(b);
-            auto data_c = get_memory<Tc>(c);
-#ifdef GGML_SYCL_NVIDIA
-            oneapi::mkl::blas::column_major::gemm(oneapi::mkl::backend_selector<oneapi::mkl::backend::cublas>{ q },
-                                                  a_trans, b_trans, m, n, k, alpha_value, data_a, lda, data_b, ldb,
-                                                  beta_value, data_c, ldc);
-#else
-            oneapi::mkl::blas::column_major::gemm(q, a_trans, b_trans, m, n, k, alpha_value, data_a, lda, data_b, ldb,
-                                                  beta_value, data_c, ldc);
-#endif
-        }
+    template <class Ta, class Tb, class Tc, class Ts>
```

Comment:
I suggest warp gemm() functions of oneMKL and oneMath in one unify API() in dpct::helper, so that make the ggml code is clear & simple.

Reply:
For Intel path, I suggest using oneMKL which comes from oneAPI official package, instead of oneMath as 3rd party.

If define a unified API, like dpct::gemm(dcpt::transpose a_trans, xxxx, int ldc), this new function will hide the difference of oneMath and oneMKL. 

That means ggml-sycl.cpp won't call oneMKL or oneMKL directly, only dpct::helper call oneMKL or oneMath.
So when we change low level code, ggml-sycl.cpp won't be impacted directly.